pkg/test/environment.go

```diff
@@ -74,9 +74,9 @@ func NewEnvironment(options ...EnvironmentOption) *Environment {
 	ctx, stop := context.WithCancel(controllerruntime.SetupSignalHandler())
 	return &Environment{
 		Environment: envtest.Environment{
-			CRDDirectoryPaths: []string{project.RelativeToRoot("config")},
+			CRDDirectoryPaths: []string{project.RelativeToRoot("config/templates/provisioner-crd.yaml")},
 			WebhookInstallOptions: envtest.WebhookInstallOptions{
-				Paths: []string{project.RelativeToRoot("config")},
+				Paths: []string{project.RelativeToRoot("config/templates/webhooks.yaml")},
```

Comment:
You should be able to just pass /config/templates to both of these.

Reply:
If I just pass config/templates to `envtest`, then the test framework will not understand helm templating tag in controller.yaml and rbac.yaml and then failed.
Another way to work around would be before ginkgo something `helm template karpenter ./config/ > /.build/manfifest.yaml`
and then in file having somthing like :
   `CRDDirectoryPaths: []string{project.RelativeToRoot("/.build/manfifest.yaml")},`
   
 What would be best the project ?

Reply:
Gotcha sounds good.